src/decoding.cc

```diff
@@ -167,13 +169,13 @@ namespace ctranslate2 {
 
       // TopK candidates.
       sampler(log_probs, topk_ids, topk_scores, _beam_size);
-      if (attention)
+      if (attention || _coverage_penalty != 0)
         attention_step.copy_from(attention_step_device);
 
       topk_log_probs = topk_scores;
       // Recover the true log probs if length penalty was applied.
       if (_length_penalty != 0)
-        ops::Mul()(topk_log_probs, StorageView(length_penalty_weight), topk_log_probs);
+        ops::Mul()(topk_log_probs, StorageView(length_penalty_weight, topk_log_probs.device()), topk_log_probs);
```

Comment:
Scalar values should be placed on CPU (the default) so this change is not required.